lib/geo_ip.rb

```diff
@@ -18,4 +14,23 @@ def country_code(ip_address)
       try(:[], 'iso_code')
   end
 
+  class << self
+    def activated?
+      if country_whitelist_configured?
+        db_present? ? true : (raise 'geo ip db file missing')
+      else
+        false
+      end
+    end
+
+    private
+
+    def db_present?
+      File.exist?(MAX_MIND_DB_PATH)
+    end
+
+    def country_whitelist_configured?
+      Setting::CountryCode.first.value.any?
```

Comment:
well, this works for now since there's only one CountryCode Setting at the moment.
Setting::CountryCode is a type.
get the desired setting by it's name !
Setting.value(:general, :country_source_whitelist)